daemon/daemon_main.go

```diff
@@ -1449,6 +1449,10 @@ func initKubeProxyReplacementOptions() {
 		log.Infof("Auto-enabling %q, %q, %q features", option.EnableNodePort,
 			option.EnableExternalIPs, option.EnableHostReachableServices)
 
+		if option.Config.Tunnel == option.TunnelDisabled {
+			option.Config.NodePortMode = option.NodePortModeHybrid
+		}
```

Comment:
Maybe we should log this somehow? :thinking:

Reply:
Yeah, makes sense.